akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Blueprint.scala

```diff
@@ -102,6 +104,7 @@ private[http] object Http2Blueprint {
       upgraded: Boolean = false): BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, NotUsed] =
     httpLayer(settings, log) atop
       serverDemux(settings.http2Settings, initialDemuxerSettings, upgraded) atop
+      serverKeepaliveIfConfigured(settings.http2Settings) atop
```

Comment:
Had some bad luck thinking here, in addition to the bidi incorrectly being reversed in the current PR state, it is trickier than just re-using the exact same logic: I tried it with a short strict request and a slow throttled streaming response, in which case the current logic keeping track if it saw any incoming data frames between pings does not work, because there will not be any incoming frames other than the pongs. Need to rethink that a bit.